core/controllers/email_dashboard_test.py

```diff
@@ -290,19 +331,29 @@ def test_that_valid_exceptions_are_raised(self):
                 query_models[0].query_status,
                 feconf.USER_QUERY_STATUS_COMPLETED)
 
-        # Check that exception is raised for unauthorized user.
-        self.login(self.USER_A_EMAIL)
-        with self.assertRaisesRegexp(Exception, '401 Unauthorized'):
-            self.get_html_response(
-                '/emaildashboardresult/%s' % query_models[0].id)
-        self.logout()
-
         # Check that exception is raised if current user is not submitter of
         # that query.
         self.login(self.NEW_SUBMITTER_EMAIL)
         with self.assertRaisesRegexp(Exception, '401 Unauthorized'):
             self.get_html_response(
                 '/emaildashboardresult/%s' % query_models[0].id)
+        # FIXME: Not catching unauthorized call for post requests.
```

Comment:
Please remove FIXME. You can login as some one and create a query model with some else's attributes I guess. That way you can test line 181. For example:
```
        self.login(self.ADMIN_EMAIL)

        query_id = user_query_services.save_new_query_model(
            self.submitter_id, inactive_in_last_n_days=10,
            created_at_least_n_exps=5,
            has_not_logged_in_for_n_days=30)
```
where submitter_id is not the user_id of ADMIN. Hope you get the gist.